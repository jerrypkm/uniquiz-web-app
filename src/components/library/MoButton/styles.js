const styles = (theme) => ({
  button: {
    boxShadow: theme.shadow?.lg,
    borderRadius: theme.space?.xs,
    "&:hover .arrow": {
      transform: "translateX(5px)",
    },
  },
  icon: {
    transition: "transform .35s",
  },
  noShadow: {
    boxShadow: "none",
  },
});

export default styles;
