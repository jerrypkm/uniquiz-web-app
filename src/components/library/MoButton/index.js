import React from "react";

import ArrowForwardIcon from "@material-ui/icons/ArrowForward";
import Button from "@material-ui/core/Button";
import { Link } from "react-router-dom";
import withStyles from "@material-ui/core/styles/withStyles";
import styles from "./styles";
import PropTypes from "prop-types";

const MoButton = ({
  classes,
  children,
  handleButtonClick,
  href,
  isArrowIcon,
  text,
  type,
  size,
  variant,
}) => {
  /* Make sure href isn't set to prevent router link errors */
  const hrefProp = href || "#";

  /* Prevent errors when handleButtonClick not available  */
  const handleOnCLick = (e) => {
    return handleButtonClick ? handleButtonClick(e) : null;
  };

  return (
    <Button
      onClick={(e) => handleOnCLick(e)}
      component={Link}
      to={hrefProp}
      type={type}
      className={`${classes.button} ${variant === "text" && classes.noShadow}`}
      color={"primary"}
      size={size}
      variant={variant}
      endIcon={
        isArrowIcon && <ArrowForwardIcon className={`${classes.icon} arrow`} />
      }
    >
      {text} {children}
    </Button>
  );
};

MoButton.propTypes = {
  classes: PropTypes.object,
  children: PropTypes.any,
  handleButtonClick: PropTypes.func,
  href: PropTypes.string,
  isArrowIcon: PropTypes.bool,
  text: PropTypes.string,
  color: PropTypes.oneOf(["primary", "secondary", "default", null]),
  size: PropTypes.oneOf(["small", "medium", "large", null]),
  type: PropTypes.oneOf(["submit", "button", null]),
  variant: PropTypes.oneOf(["contained", "outlined", "text", null]),
};

export default withStyles(styles)(MoButton);
