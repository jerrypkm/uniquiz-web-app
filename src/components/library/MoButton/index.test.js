import React from "react";

import ShallowRenderer from "react-test-renderer/shallow";
import MoButton from "./index.js";

const renderer = new ShallowRenderer();

describe("MoButton Component", () => {
  test("should match the snapshot", () => {
    renderer.render(<MoButton />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
