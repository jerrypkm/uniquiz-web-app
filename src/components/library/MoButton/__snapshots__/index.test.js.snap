// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoButton Component should match the snapshot 1`] = `
<MoButton
  classes={
    Object {
      "button": "MoButton-button-1",
      "icon": "MoButton-icon-2",
      "noShadow": "MoButton-noShadow-3",
    }
  }
/>
`;
